packages/abstractions/src/Authentication/AccessTokenProvider.php

```diff
@@ -8,7 +8,8 @@
 
 namespace Microsoft\Kiota\Abstractions\Authentication;
 
-use Http\Promise\Promise;
+use Microsoft\Kiota\Http\Promise;
```

Comment:
I don't think the authentication package should have a dependency on the HTTP package. It effectively locks in the both when the goal of having two separate packages was to enable people using them independently. 
Maybe the premise definitions should be moved to the abstractions package?

Reply:
Sure, no problem. I just need to confirm this proof of concept, as this is the way to go?

Reply:
So I pushed new commit, which changed the namespace to the Microsoft\Kiota\Abstraction\Promise